pysb/simulator/cupsoda.py

```diff
@@ -462,8 +463,11 @@ def _get_cmatrix(self):
         par_vals = self.param_values[:, rate_mask]
         rate_order = []
         for rxn in self._model.reactions:
-            rate_args.append([arg for arg in rxn['rate'].args if
-                              not re.match("_*s", str(arg))])
+            if isinstance(rxn['rate'], pysb.Component):
```

Comment:
I think this if/else could be eliminated by using more of the sympy API:
```python
rate_args.append([arg for arg in rxn['rate'].atoms(sympy.Symbol) if not arg.name.startswith('__s')])
```
(I also made the assumption that `.startswith('__s')` is what you are actually after here to match the species symbols, rather than that `_*s` regex)

Reply:
That works -- thanks.